controller_manager/test/test_hardware_resources/test_sensor_interface.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <vector>
+
+#include "./test_sensor_interface.hpp"
+
+hardware_interface::return_type
+TestSensor::configure(const hardware_interface::HardwareInfo & sensor_info)
+{
+  fprintf(stderr, "configuring plugin with name %s\n", sensor_info.name.c_str());
+  return hardware_interface::return_type::OK;
+}
+
+hardware_interface::return_type
+TestSensor::start()
+{
+  return hardware_interface::return_type::OK;
+}
+
+hardware_interface::return_type
+TestSensor::stop()
+{
+  return hardware_interface::return_type::OK;
+}
+
+hardware_interface::hardware_interface_status
+TestSensor::get_status() const
+{
+  return hardware_interface::hardware_interface_status::UNKNOWN;
+}
+
+hardware_interface::return_type
+TestSensor::read_sensors(
```

Comment:
@destogl while implementing this, I was wondering why a sensor has to implement multiple sensors? Why not only a single component?